keyvault/client/src/main/java/com/azure/keyvault/SecretClient.java

```diff
@@ -0,0 +1,540 @@
+     * <p>
+     * The name and value fields in {@code secret} are required.
+     * The expires, contentType and and notBefore values value for the Secret are optional.
+     * If not specified, no values are set for the fields.
+     * </p>
+     *
+     * @param secret The Secret object containing information about the secret and its properties.
+     * @throws NullPointerException if {@code secret} is {@code null}.
+     * @return the Mono to the Secret object
+     */
+    public Mono<RestResponse<Secret>> setSecretAsync(Secret secret) {
+        Objects.requireNonNull(secret, "The Secret input parameter cannot be null.");
+        Objects.requireNonNull(secret.name(), "The Secret name cannot be null.");
+        Objects.requireNonNull(secret.value(), "The Secret value cannot be null.");
+
+        SecretRequestAttributes secretRequestAttributes = new SecretRequestAttributes()
+                                                .withEnabled(secret.enabled())
+                                                .withExpires(secret.expires())
+                                                .withNotBefore(secret.notBefore());
+
+        SecretRequestOptions parameters = new SecretRequestOptions()
+                                            .withValue(secret.value())
+                                            .withTags(secret.tags())
+                                            .withContentType(secret.contentType())
+                                            .withSecretAttributes(secretRequestAttributes);
+
+        return service.setSecret(vaultEndPoint, secret.name(), API_VERSION,ACCEPT_LANGUAGE , parameters);
+    }
+
+
+    /**
+     * Sets a secret in the key vault.
+     * The SET operation adds a secret to the Azure Key Vault.
+     * If the named secret already exists, Azure Key Vault creates a new version of that secret.
+     * This operation requires the secrets/set permission.
+     *
+     * @param name The name of the secret. This field
+     * @param value The value of the secret.
+     * @throws NullPointerException if {@code name} or {@code value} parameter is {@code null}.
+     * * @return the Secret that was created.
+     */
+    public Mono<RestResponse<Secret>> setSecretAsync(String name, String value) {
+        Objects.requireNonNull(name, "The Secret name cannot be null.");
+        Objects.requireNonNull(value, "The Secret value cannot be null.");
+
+        SecretRequestOptions parameters = new SecretRequestOptions()
+                                            .withValue(value);
+        return service.setSecret(vaultEndPoint, name, API_VERSION, ACCEPT_LANGUAGE, parameters);
+    }
+
+
+    /**
+     * Get a specified secret from the key vault.
+     * The GET operation is applicable to any secret stored in Azure Key Vault.
+     * This operation requires the secrets/get permission.
+     *
+     * @param name The name of the secret, cannot be null
+     * @param version The version of the secret, cannot be null
```

Comment:
This seems more demanding than it needs to be - but it might be a Key Vault thing. For example, rather than demanding `version` be non-null, couldn't you alternatively call the `getSecretAsync(String name)` method if `version` is null?

Reply:
Updated, 
if version is null. latest version is retrieved.